src/encoder.c

```diff
@@ -665,3 +688,115 @@ const OSSL_DISPATCH p11prov_ec_encoder_spki_der_functions[] = {
+    if (p) {
+        ret = p11prov_print_buf(out, p, "Pub:", 4);
+        if (ret != RET_OSSL_OK) {
+            return RET_OSSL_ERR;
+        }
+    }
+
+    /* Name */
+    p = OSSL_PARAM_locate_const(params, OSSL_PKEY_PARAM_GROUP_NAME);
+    if (p) {
+        const char *name;
+        int nid;
+
+        ret = OSSL_PARAM_get_utf8_string_ptr(p, &name);
+        if (ret != RET_OSSL_OK) {
+            return RET_OSSL_ERR;
+        }
+
+        BIO_printf(out, "ASN1 OID: %s\n", name);
+
+        /* Print also NIST name if any */
+        nid = OBJ_txt2nid(name);
+        if (nid != NID_undef) {
+            name = EC_curve_nid2nist(nid);
+            if (name) {
+                BIO_printf(out, "NIST CURVE: %s\n", name);
+            }
+        }
+    }
+
+    return RET_OSSL_OK;
+}
+
+static int p11prov_ec_encoder_encode_text(void *inctx, OSSL_CORE_BIO *cbio,
+                                          const void *inkey,
+                                          const OSSL_PARAM key_abstract[],
+                                          int selection,
+                                          OSSL_PASSPHRASE_CALLBACK *cb,
+                                          void *cbarg)
+{
+    struct p11prov_encoder_ctx *ctx = (struct p11prov_encoder_ctx *)inctx;
+    P11PROV_OBJ *key = (P11PROV_OBJ *)inkey;
+    CK_KEY_TYPE type;
+    CK_ULONG keysize;
+    CK_ATTRIBUTE *a;
+    BIO *out;
+    int ret;
+
+    P11PROV_debug("EC Text Encoder");
+
+    type = p11prov_obj_get_key_type(key);
+    if (type != CKK_EC) {
+        P11PROV_raise(ctx->provctx, CKR_GENERAL_ERROR, "Invalid Key Type");
+        return RET_OSSL_ERR;
+    }
+
+    out = BIO_new_from_core_bio(p11prov_ctx_get_libctx(ctx->provctx), cbio);
+    if (!out) {
+        P11PROV_raise(ctx->provctx, CKR_GENERAL_ERROR, "Failed to init BIO");
+        return RET_OSSL_ERR;
+    }
+
+    keysize = p11prov_obj_get_key_bit_size(key);
+    if (selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) {
+        CK_OBJECT_CLASS class = p11prov_obj_get_class(key);
+        if (class != CKO_PRIVATE_KEY) {
+            return RET_OSSL_ERR;
+        }
+        BIO_printf(out, "PKCS11 EC Private Key (%lu bits)\n", keysize);
+        BIO_printf(out, "[Can't export and print private key data]\n");
+    }
+
+    if (selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) {
+        BIO_printf(out, "PKCS11 EC Public Key (%lu bits)\n", keysize);
+        ret = p11prov_obj_export_public_ec_key(key, p11prov_ec_print_public_key,
+                                               out);
+        if (ret != RET_OSSL_OK) {
+            BIO_printf(out, "[Error: Failed to decode public key data]\n");
+        }
+    }
+
+    a = p11prov_obj_get_attr(key, CKA_ID);
+    if (a) {
+        BIO_printf(out, "Key ID:\n");
+        ASN1_buf_print(out, a->pValue, a->ulValueLen, 4);
+    }
+    a = p11prov_obj_get_attr(key, CKA_LABEL);
+    if (a) {
+        BIO_printf(out, "Label: %*s\n", (int)a->ulValueLen, (char *)a->pValue);
+    }
```

Comment:
just a nit that this part is the same as for the rsa so this might be an opportunity for helper function to avoid code duplication, but I do not insist on this one.

Reply:
Yes, I was almost going to rehaul the whole code, but then scrapped it to keep this PR "small".

Once we get this one merged I may open a new PR with refactoring of the whole thing, as I thought of a way to simplify some of the code when allocations are needed.